### DISCUSSION TITLE
"fedoraremix.exe -s -c tmux" not working

### DISCUSSION THREAD

**Discussion by returntrip** [WhitewaterFoundry/Fedora-Remix-for-WSL#183] [Q&A] [answered]
Should I be able to use -s and -c <command> at the same time?

**Comment by crramirez** [comment, marked as answer]
Hello @returntrip,

For now, once you use the **-s** parameter, you cannot send anything else.

You can, for example add the `tmux` at the end of `.bashrc`.

Another option is to use the new WSL built-in SystemD adding to the /etc/wsl.conf 

```ini
[boot]
systemd=true
```
In this case, you can call `fedoraremix.exe -c tmux`

**Reply by returntrip** [reply]
Thanks for your reply.

Is there aany documentation about pro and cons of using WSL's built in systemd vs the Fedora WSL remix version?

**Reply by crramirez** [reply]
No, the only thing, for now, is that the Gnome desktop via remote desktop, as described in the blog post, is not working with built-in SystemD. Everything else works fine. Snaps applications with WSLg work better with the built-in SystemD.

Apart from that, the built-in may be activated or deactivated, whereas the fedora one you can use or not depending on what you need at the time.

Eventually, we will integrate both as we did with Pengwin. So it will be transparent for you which method is being used depending on the WSL version installed. Of course, when all the issues had been resolved.

**Comment by returntrip** [comment]
I have tried enabling built-in systemd but it does not work (I guess because am running Windows version 10.0.19045.2251, the joys of corporate environments...).

```
$ cat /etc/wsl.conf
[automount]
enabled = true
options = "metadata,uid=1000,gid=1000,umask=22,fmask=11,case=off"
mountFsTab = true

[boot]
systemd=true

[network]
generateHosts = true
generateResolvConf = true


[user]
default=user

$ systemctl status pcscd.service
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
$ systemctl status
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
```

**Reply by crramirez** [reply]
Check that you have this update https://support.microsoft.com/en-us/topic/november-15-2022-kb5020030-os-builds-19042-2311-19043-2311-19044-2311-and-19045-2311-preview-237a9048-f853-4e29-a3a2-62efdbea95e2 

After that, run: `wsl --update --web-download`

For more details: https://devblogs.microsoft.com/commandline/the-windows-subsystem-for-linux-in-the-microsoft-store-is-now-generally-available-on-windows-10-and-11/

**Reply by returntrip** [reply]
Fantastic. That update was necessary! Now systemd works with the built in version. Many thanks!
